### ISSUE TITLE
Document how to inject styles into components not implementing ThemableMixin from server-side

### ISSUE THREAD

**Issue by mvysny** [vaadin/docs#790]
Document how to inject styles into components not implementing ThemableMixin from server-side.

Currently the page at https://vaadin.com/docs/v14/flow/styling/importing-style-sheets only describe how to style components that use ThemableMixin, including all Vaadin components. However there is no documentation on how to style third-party external web components (like https://vaadin.com/directory/component/full-calendar-flow ).

Also see https://vaadin.com/forum/thread/17827990/where-to-include-a-web-components-css-file-when-integrating-a-third-party-w

**Comment by mvysny** [comment]
Workaround: extend the class in javascript and add the styles manually, as described at https://vaadin.com/directory/component/full-calendar-flow/samples , the "using a custom class" example.

**Comment by mvysny** [comment]
Another workaround: simply inject new `<style>` element into shadow-dom as follows:
```java
component.getElement().executeJs("var s = document.createElement(\"style\"); s.textContent = \".fc-timegrid-slots { background: red }\"; this.shadowRoot.appendChild(s);");
```

**Comment by jouni** [comment]
> Apparently those only works for components using `ThemableMixin` but none of web components outside of Vaadin components is using that stuff since it's Vaadin-proprietary.

Yeah, the [Component Styles](https://vaadin.com/docs/v14/flow/styling/importing-style-sheets/#component-styles) section states this in the beginning.

**Comment by jouni** [comment]
> Document how to inject styles into components not implementing ThemableMixin from server-side.

This sounds like a good Cookbook recipe as well.

**Comment by jouni** [comment]
I believe this is not something we should document. If a 3rd-party component doesn't offer a styling API, it’s outside our control. Vaadin component’s are also moving away from ThemableMixin and injecting custom styles into the shadow root of the components with ::part-based styling, which makes it a bit more questionable why we would document that for 3rd-party components.

**Comment by mvysny** [comment]
I understand that we can't provide documentation for 3rd party components. However, please consider this ticket: https://github.com/vaadin/flow/issues/19067 - it's a Vaadin Directory component which I'm trying to style. CssImport doesn't work and doesn't tell me why it doesn't work. The documentation doesn't say anything either, it doesn't even mention the `ThemableMixin`. I, as a Vaadin user, am completely lost.

I,as a Vaadin developer, would definitely welcome this:
* if the documentation would mention that CssImport only works with components implementing ThemableMixin;
* if the webcomponent doesn't offer any themable parts nor implements ThemableMixin the documentation could mention the abovementioned workarounds or link cookbook examples.

**Comment by mvysny** [comment]
> > Apparently those only works for components using `ThemableMixin` but none of web components outside of Vaadin components is using that stuff since it's Vaadin-proprietary.
> 
> Yeah, the [Component Styles](https://vaadin.com/docs/v14/flow/styling/importing-style-sheets/#component-styles) section states this in the beginning.

True, but only for Vaadin 14. The Vaadin 24 documentation doesn't document this at all unfortunately.
